tests/utils/v2Data/events.js

```diff
@@ -0,0 +1,311 @@
+const uuid = require('uuid');
+const { UtapiMetric } = require('../../../libV2/models');
+
+const { range, randChoice, randInt } = require('./rand');
+
+const requiredFields = {
+    uuid: () => uuid.v4(),
+    account: params => params.account,
+    location: params => params.location || 'ring',
+};
+
+const objOpFields = {
+    user: params => params.user,
+    bucket: params => params.bucket,
+    object: params => params.object,
+};
+
+const eventTemplates = {
+    get: {
+        ...objOpFields,
+        outgoingBytes: params => params.currentSize,
+    },
+    put: {
+        ...objOpFields,
+        incomingBytes: params => params.newSize,
+        sizeDelta: params => {
+            if (params.versioning) {
+                return params.newSize;
+            }
+            return params.newSize - params.currentSize;
+        },
+        objectDelta: params => {
+            if (params.versioning) {
+                return 1;
+            }
+            return params.exists ? 0 : 1;
+        },
+    },
+    delete: {
+        ...objOpFields,
+        sizeDelta: params => {
+            if (params.versioning) {
+                return 0;
+            }
+            return Math.floor((params.currentSize / params.versions)) * -1;
+        },
+        objectDelta: params => (params.versioning ? 0 : -1),
+    },
+    multidelete: {
+        ...objOpFields,
+        sizeDelta: params => {
+            if (params.versioning) {
+                return 0;
+            }
+            return Math.floor((params.currentSize / params.versions)) * -2;
+        },
+        objectDelta: params => (params.versioning ? 0 : -2),
+        object: () => undefined,
+    },
+    md: {
+        user: params => params.user,
+    },
+
+    bucket: {
+        user: params => params.user,
+        bucket: params => params.bucket,
+    },
+    obj: objOpFields,
+};
+
+const eventTemplateToOperation = {
+    get: [
+        'getObject',
+    ],
+
+    put: [
+        'copyObject',
+        'putData',
+        'putObject',
+        'uploadPart',
+        'uploadPartCopy',
+    ],
+
+    delete: [
+        'deleteObject',
+    ],
+
+    multidelete: [
+        'multiObjectDelete',
+    ],
+
+    bucket: [
+        'abortMultipartUpload',
+        'completeMultipartUpload',
+        'initiateMultipartUpload',
+        'listMultipartUploadParts',
+        'listMultipartUploads',
+        'createBucket',
+        'deleteBucket',
+        'deleteBucketCors',
+        'deleteBucketReplication',
+        'deleteBucketReplication',
```

Comment:
same here `deleteBucketReplication` is twice. I am now thinking if its at 2 places, is there a reason behind it?

Reply:
copy/paste

Reply:
are you copy/pasting the thumbs up? xD